### DISCUSSION TITLE
How to add an ExtractionChain?

### DISCUSSION THREAD

**Discussion by scorego** [FlowiseAI/Flowise#735] [Q&A]
I want to use extraction chain with Flowise. Extraction chain is implemented on `LLMChain`, with customized `output_parser` and `llm_kwargs`, and I cannot find how to configure these parameters in Flowise UI. Is there any plan to support customizing them?
![image](https://github.com/FlowiseAI/Flowise/assets/10768254/f7037adf-ae3f-4557-8f93-bb030292e341)

SEE https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/chains/openai_functions/extraction.py#L46
```
def create_extraction_chain(
    schema: dict, llm: BaseLanguageModel, verbose: bool = False
) -> Chain:

    function = _get_extraction_function(schema)
    prompt = ChatPromptTemplate.from_template(_EXTRACTION_TEMPLATE)
    output_parser = JsonKeyOutputFunctionsParser(key_name="info")
    llm_kwargs = get_llm_kwargs(function)
    chain = LLMChain(
        llm=llm,
        prompt=prompt,
        llm_kwargs=llm_kwargs,
        output_parser=output_parser,
        verbose=verbose,
    )
    return chain
```

PS, I think UI should provide a way to customize all the class field of LLMChain.
![image](https://github.com/FlowiseAI/Flowise/assets/10768254/e1252a9c-a477-48dd-ad99-7eaa14d3b102)

**Comment by chungyau97** [comment]
Hi @scorego,

Would you like to take up this part to submit a PR?
Here's the source code [location](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/chains/LLMChain/LLMChain.ts#L67)
